scripts/add-ocp-labels/main.go

```diff
@@ -0,0 +1,634 @@
+package main
+
+import (
+	goerror "errors"
+	"fmt"
+	"github.com/blang/semver"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+
+	apimanifests "github.com/operator-framework/api/pkg/manifests"
+	apivalidation "github.com/operator-framework/api/pkg/validation"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const ocpLabelAnnotation = "com.redhat.openshift.versions"
+
+// OCP version where the apis v1beta1 is no longer supported
+const ocpVerV1beta1Unsupported = "4.9"
+
+type BundleAnnotations struct {
+	Annotations map[string]string `yaml:"annotations,omitempty"`
+}
+
+type UpdateGraph struct {
+	UpdateGraph string `yaml:"annotations,omitempty"`
+}
+
+type BundleDir struct {
+	Path  string
+	Error string
+}
+
+type Output struct {
+	Errors                       []BundleDir
+	RequiresUpdate               []BundleDir
```

Comment:
suggestion: Some of these read like they might be booleans. E.g. `RequiresUpdatePkgFormat`. Maybe, even though it does make it longer, something like `BundlesRequiringPkgFormatUpdate` might be more readable?